examples/jsm/nodes/accessors/MaterialNode.js

```diff
@@ -135,6 +169,134 @@ class MaterialNode extends Node {
 
 			node = new MaterialReferenceNode( 'rotation', 'float' );
 
+		} else if ( scope === MaterialNode.UV ) {
+
+			// uv repeat and offset setting priorities
+			// 1. color map
+			// 2. specular map
+			// 3. displacementMap map
+			// 4. normal map
+			// 5. bump map
+			// 6. roughnessMap map
+			// 7. metalnessMap map
+			// 8. alphaMap map
+			// 9. emissiveMap map
+			// 10. clearcoat map
+			// 11. clearcoat normal map
+			// 12. clearcoat roughnessMap map
+			// 13. iridescence map
+			// 14. iridescence thickness map
+			// 15. specular intensity map
+			// 16. specular tint map
+			// 17. transmission map
+			// 18. thickness map
+
+			let uvScaleMap;
+			let uvNode;
+
+			if ( material.map ) {
```

Comment:
Maybe it is better to write this as `uvScaleMap = material.map || material.specularMap || ...`?

Reply:
Interesting... maybe we should do this in core too?